lib/taskana-core/src/test/java/acceptance/classification/CreateClassificationAccTest.java

```diff
@@ -1,297 +1,374 @@
-    assertThat(classification.getServiceLevel()).isEqualTo("P0D");
+  @WithAccessId(user = "businessadmin")
+  @AfterAll
+  void should_NotCreateClassification_When_ExceptionIsThrownDuringCreation() {
+    List<ClassificationSummary> classifications =
+        classificationService.createClassificationQuery().keyIn("KeyErrCreation").list();
 
-    classification.setServiceLevel("");
-    classification = CLASSIFICATION_SERVICE.updateClassification(classification);
-    assertThat(classification.getServiceLevel()).isEqualTo("P0D");
+    assertThat(classifications).isEmpty();
+  }
+
+  @Nested
+  @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+  class CreatingChildClassificationTest {
+    String parentId;
+    String parentKey;
+
+    @WithAccessId(user = "businessadmin")
+    @BeforeAll
+    void setupParentClassification() throws Exception {
+      ClassificationSummary parentClassification =
+          DefaultTestEntities.defaultTestClassification()
+              .key("Key0")
+              .domain("DOMAIN_A")
+              .type("TASK")
+              .buildAndStoreAsSummary(classificationService);
+      parentId = parentClassification.getParentId();
+      parentKey = parentClassification.getParentKey();
+    }
+
+    @WithAccessId(user = "businessadmin")
+    @Test
+    void should_CreateClassification_WhenCreatingChildClassification() throws Exception {
+      Classification childClassification =
+          classificationService.newClassification("Key0_1", "DOMAIN_A", "TASK");
+      childClassification.setParentId(parentId);
+      childClassification.setParentKey(parentKey);
+
+      assertThatCode(() -> classificationService.createClassification(childClassification))
+          .doesNotThrowAnyException();
+      List<ClassificationSummary> classifications =
+          classificationService.createClassificationQuery().parentIdIn(parentId).list();
+      assertThat(classifications).contains(childClassification.asSummary());
+    }
+
+    @WithAccessId(user = "businessadmin")
+    @Test
+    void should_CreateMasterClassification_WhenCreatingChildClassification() throws Exception {
+      Classification childClassification =
+          classificationService.newClassification("Key0_2", "DOMAIN_A", "TASK");
+      childClassification.setParentId(parentId);
+      childClassification.setParentKey(parentKey);
+      classificationService.createClassification(childClassification);
+
+      ClassificationImpl expectedMasterClassification =
+          (ClassificationImpl) childClassification.copy("Key0_2");
+      expectedMasterClassification.setDomain("");
+      expectedMasterClassification.setParentId(parentId);
+      expectedMasterClassification.setParentKey(parentKey);
+      List<ClassificationSummary> classifications =
+          classificationService.createClassificationQuery().keyIn("Key0_2").list();
+      assertThat(classifications)
+          .usingRecursiveFieldByFieldElementComparatorIgnoringFields("id")
```

Comment:
should we verify that an id exists?

Reply:
I think, we could do that with a idIn-Parameter in the Query, right? So, it's minimal effort therefore we could verify that too.